beacon-chain/sync/rpc_status.go

```diff
@@ -73,8 +73,12 @@ func (s *Service) resyncIfBehind() {
 	runutil.RunEvery(s.ctx, interval, func() {
 		if s.shouldReSync() {
 			syncedEpoch := helpers.SlotToEpoch(s.chain.HeadSlot())
-			highestEpoch, _ := s.p2p.Peers().BestNonFinalized(flags.Get().MinimumSyncPeers, syncedEpoch)
-			if helpers.StartSlot(highestEpoch) > s.chain.HeadSlot() {
+			// Factor number of expected minimum sync peers, to make sure that enough peers are
+			// available to resync (some peers may go away between checking non-finalized peers and
+			// actual resyncing).
+			highestEpoch, _ := s.p2p.Peers().BestNonFinalized(flags.Get().MinimumSyncPeers*2, syncedEpoch)
+			// Check if the current node is more than 1 epoch behind.
+			if (highestEpoch - 1) > syncedEpoch {
```

Comment:
This will overflow if `highestEpoch` is 0

Reply:
1) will fix the overflow now
2) From discord:
> Nishant, regarding your comment (not the overflow one, but above), how come BestNonFinalized requires 30 peers now? Best non-finalized before took just minimum peers to sync (3 by default), but when peers are blinking, it posed problem, so doubling (to 6) allows some leeway.
I am not sure if I understand concerns in your comment.
The problem we had is that it was "too easy" to cause resync or update to queue's highest finalized slot, now with double of min sync peers (3*2=6 by default), requirements a bit more hard, which should allow for a more robust system.
Everywhere except for those two places, we are still quite ok with min.number of peers (3 by default), and sync progresses. So, the only affected parts are: 1) "resync trigger" and 2) "force stay in queue", and it takes a bit more peers now to convince the system that it is the right move
Am I missing sth here?